lib/public/ConfigLexicon/ConfigLexiconEntry.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2024 Maxence Lange <maxence@artificial-owl.com>
+ *
+ * @author Maxence Lange <maxence@artificial-owl.com>
+ *
+ * @license AGPL-3.0 or later
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCP\ConfigLexicon;
+
+/**
+ * Model that represent config values within an app config lexicon.
+ *
+ * @see IConfigLexicon
+ * @since 30.0.0
+ */
+class ConfigLexiconEntry implements IConfigLexiconEntry {
+	private string $definition = '';
+	private ?string $default = null;
+
+	/**
+	 * @param string $key config key
+	 * @param int $valueType type of config value ({@see self::TYPE_STRING} and others)
+	 * @param string $definition optional description of config key available when using occ command
+	 * @param bool $lazy set config value as lazy
+	 * @param bool $sensitive set config value as sensitive
+	 * @param bool $deprecated set config key as deprecated
+	 * @since 30.0.0
+	 */
+	public function __construct(
+		private readonly string $key,
+		private readonly int $valueType,
+		string $definition = '',
+		private readonly bool $lazy = false,
+		private readonly bool $sensitive = false,
+		private readonly bool $deprecated = false
```

Comment:
```suggestion
		private ?string $default = null,
		private readonly bool $lazy = false,
		private readonly bool $sensitive = false,
		private readonly bool $deprecated = false,
		string $definition = '',
```

Can add the default here too. It's mostlikely the most used option so would put it to the front of the list as well.

Reply:
while the code convert the `$default` value to string (the same way it is stored in the database) forcing the string type here might generate conflicts. using method like `withDefault<type>()` ensure the dev set a correctly typed value

Reply:
the constructor could call it as type is given as well?